kitsune/wiki/tests/test_templates.py

```diff
@@ -1013,6 +1013,143 @@ def test_keywords_require_permission(self):
+        cache.clear()
+        trans_resp = self.client.get(trans_url)
+        trans_content = pq(trans_resp.content)
+        # Check user message is shown there
+        eq_(1, len(trans_content('.user-messages li')))
+        # Check there are two buttons for restore and discard
+        eq_(2, len(trans_content('.user-messages .info form .btn')))
+        # Restore with the draft data
+        draft_request = {'restore': 'Restore'}
+        cache.clear()
+        trans_resp = self.client.get(trans_url, draft_request)
+        trans_content = pq(trans_resp.content)
+        # No user message is shown there
+        eq_(0, len(trans_content('.user-messages li')))
+        # Check title, slug, keywords, content etc are restored
+        restored_data = {'title': trans_content('#id_title').val(),
+                         'slug': trans_content('#id_slug').val(),
+                         'keywords': trans_content('#id_keywords').val(),
+                         'summary': trans_content('#id_summary').text(),
+                         'content': trans_content('#id_content').text(),
+                         'based_on': int(trans_content('#id_based_on').val())}
+        for key in data:
+            eq_(data[key], restored_data[key])
+
+    def test_restore_draft_revision_with_older_based_on(self):
+        """Test restoring a draft which is based on an old based on"""
+        rev1 = ApprovedRevisionFactory(is_ready_for_localization=True)
+        doc = rev1.document
+        doc.is_localizable = True
+        doc.save()
+        data = {
+            'content': 'Test Content bla bla bla',
+            'keywords': 'something',
+            'summary': 'some summary',
+            'title': 'test title',
+            'slug': 'test-slug',
+            'based_on': int(rev1.id)}
+        # Make a Draft with above mentioned data for bn-BD locale
+        draft_url = reverse('wiki.draft_revision', locale='bn-BD')
+        draft_resp = self.client.post(draft_url, data)
+        eq_(201, draft_resp.status_code)
+        # Create another revision in the parent doc
+        rev2 = ApprovedRevisionFactory(document=doc, is_ready_for_localization=True)
+        # Check this rev2 revision content is the translation page
+        trans_url = reverse('wiki.translate', locale='bn-BD', args=[doc.slug])
+        trans_resp = self.client.get(trans_url)
+        trans_content = pq(trans_resp.content)
+        eq_(rev2.content, trans_content('.content textarea').text())
+        # Now Restore the draft which is based on the past revision
+        draft_request = {'restore': 'Restore'}
+        cache.clear()
+        trans_resp = self.client.get(trans_url, draft_request)
+        trans_content = pq(trans_resp.content)
+        eq_(rev1.content, trans_content('.content textarea').text())
+        # As the old based on draft is restored, the latest revision content should not be there
+        assert not rev2.content == trans_content('.content textarea').text()
```

Comment:
you probably want `assert rev2.content != trans_content('.content textarea').text()`. That's a bit clearer.